test/e2e/artifacts_test.go

```diff
@@ -818,6 +818,52 @@ spec:
 		WaitForWorkflow(fixtures.ToBeSucceeded)
 }
 
+// TestDeleteArtifactGCWorkflow test no artifact generated in the workflow, the artifact gc should be successful
+// and the finalizer should be removed
+func (s *ArtifactsSuite) TestDeleteArtifactGCWorkflow() {
+	s.Given().
+		Workflow(`apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: artifact-gc-
+spec:
+  entrypoint: main
+  artifactGC:
+    strategy: OnWorkflowDeletion
+  templates:
+    - name: main
+      container:
+        image: argoproj/argosay:v2
+        command:
+          - sh
+          - -c
+        args:
+          - |
+            exit 1
+      outputs:
+        artifacts:
+          - name: on-completion
+            path: /tmp/on-completion.txt
+            s3:
+              key: on-completion.txt
+            artifactGC:
+              strategy: OnWorkflowCompletion
+          - name: on-deletion
+            path: /tmp/on-deletion.txt
+            s3:
+              key: on-deletion.txt`).
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(fixtures.ToBeFailed).
+		Then().
+		ExpectWorkflow(func(t *testing.T, meta *metav1.ObjectMeta, _ *wfv1.WorkflowStatus) {
+			assert.Equal(t, 0, len(meta.Finalizers))
+		}).
+		When().
+		DeleteWorkflow().
+		WaitForWorkflowDeletion()
+}
```

Comment:
did it work better to add the separate test as opposed to incorporating it as a case in `TestArtifactGC()`?